index.src.html

```diff
@@ -214,15 +256,48 @@
     <a>control messages</a> in the <a>control message queue</a>, and all pending
     callbacks.
 
-    When invoked, run the <a>Reset</a> algorithm.
+    When invoked, run the <a>Reset AudioDecoder</a> algorithm.
   </dd>
 
   <dt><dfn method for=AudioDecoder>close()</df></dt>
   <dd>
     Immediately aborts all pending work and releases system resources. Close is
     permanent.
 
-    When invoked, run the <a>Close</a> algorithm.
+    When invoked, run the <a>Close AudioDecoder</a> algorithm.
+  </dd>
+</dl>
+
+Algorithms {#audiodecoder-algorithms}
+-------------------------------------
+<dl>
+  <dt><dfn>Output AudioFrame</dfn> (with |output|)</dt>
+  <dd>
+    Run these steps:
+    1. Let |buffer| be an {{AudioBuffer}} containing the decoded audio data in
+        |output|.
+    2. Let |frame| be an {{AudioFrame}} containing |buffer| and a timestamp for
+        the output.
+    3. Invoke {{AudioDecoder/[[output callback]]}} with frame.
```

Comment:
This algo is always run in an async task. LMK if you think that should be moved from the caller to the callee.

Reply:
Is that new? I was thinking we could follow WebAudio with lines like 'Queue a task to invoke errorCallback with error.'

https://www.w3.org/TR/webaudio/#ref-for-dom-baseaudiocontext-decodeaudiodata%E2%91%A0

Reply:
This is slightly different, because this queues a task to invoke a WebIDL callback on the control thread (which is always the main thread for now). Here, we're just calling a callback from an operation. It happens, that, for now this operation is always called from a thread that is set up to invoke a callback, but that's just the way it is now.

Additionally, queuing to a media element task source doesn't really work because it's really possible to:
- not use any `HTMLMediaElement` for any of this, and the [media element task source is the task source of a media element](https://html.spec.whatwg.org/multipage/media.html#media-element-event-task-source), and we don't have one, so we don't have a task source
- even if we had an `HTMLMediaElement`, all this is intended to usable from a web worker, and we can't have an `HTMLMediaElement` in a worker (it's main thread-only)

I think it would be best to have an abstract "control thread", which is the thread that is being used to schedule web codecs operation (be it a web worker thread or the main thread). This thread will have an event loop, and we can dispatch tasks to it, and everything will be extra clear and have all the properties we want.

(We do the same in the Web Audio API, because even though it's main-thread only today, numerous authors have requested web worker access, and it felt best to have this layer of indirection for when we spec it, to not have to change the spec everywhere once we allow Web Audio API usage from a Web Worker).